gef-ui/server/controllers/amendments/facility-end-date/get-facility-end-date.ts

```diff
@@ -0,0 +1,77 @@
+import { CustomExpressRequest, DayMonthYearInput } from '@ukef/dtfs2-common';
+import { format } from 'date-fns';
+import { Response } from 'express';
+import * as api from '../../../services/api';
+import { FacilityEndDateViewModel } from '../../../types/view-models/amendments/facility-end-date-view-model';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { userCanAmendFacility } from '../../../utils/facility-amendments.helper';
+import { getPreviousPage } from '../helpers/navigation.helper';
+import { PORTAL_AMENDMENT_PAGES } from '../../../constants/amendments';
+
+export type GetFacilityEndDateRequest = CustomExpressRequest<{
+  params: { dealId: string; facilityId: string; amendmentId: string };
+}>;
+
+/**
+ * Controller to get the `Facility end date` page
+ * @param req - the request object
+ * @param res - the response object
+ */
+export const getFacilityEndDate = async (req: GetFacilityEndDateRequest, res: Response) => {
+  try {
+    const { dealId, facilityId, amendmentId } = req.params;
+    const { userToken, user } = asLoggedInUserSession(req.session);
+
+    const deal = await api.getApplication({ dealId, userToken });
+    const { details: facility } = await api.getFacility({ facilityId, userToken });
+
+    if (!deal || !facility) {
+      console.error('Deal %s or Facility %s was not found', dealId, facilityId);
+      return res.redirect('/not-found');
+    }
+
+    if (!userCanAmendFacility(facility, deal, user.roles)) {
+      console.error('User cannot amend facility %s on deal %s', facilityId, dealId);
+      return res.redirect(`/gef/application-details/${dealId}`);
+    }
+
+    const amendment = await api.getAmendment({ facilityId, amendmentId, userToken });
+
+    if (!amendment) {
+      console.error('Amendment %s not found on facility %s', amendmentId, facilityId);
```

Comment:
```suggestion
      console.error('Amendment %s is not found on the facility %s', amendmentId, facilityId);
```

Could you please also update other functions with similar message please.